### DISCUSSION TITLE
Applying an OpenMapTiles-based style to vector tiles generated with Tilemaker

### DISCUSSION THREAD

**Discussion by diehl** [systemed/tilemaker#729] [Q&A] [answered]
Last night, I built PMTiles for North America from the corresponding osm.pbf file for the region. Then, I tried applying a custom style that I developed with Maptiler which uses OpenMapTiles as the underlying data source. Some of the underlying map data is not rendering straightaway. Should I, in theory, be able to take an existing style for a map using OpenMapTiles data and apply it without issue to vector tiles built with Tilemaker?

**Comment by systemed** [comment]
Can you say what those differences are?

The c.t OMT-compatible schema is effectively a best-effort reimplementation and is not 1:1 in all areas. Much of OMT is written with a SQL database in mind which is not something tilemaker can recreate. There are also some ambiguities in the OMT spec, particularly around place/street names.

**Reply by diehl** [reply]
@systemed What is most obvious is that place names are missing along with state boundaries in CONUS. 

A related question that is coming up for me is whether it's possible to programmatically identify mismatches between a style JSON spec and a set of vector tiles. Wonder if anything comes to mind to accomplish this?

**Reply by diehl** [reply]
I'd love to be able to leverage Tilemaker since it seems way faster and less resource intensive than the OpenMapTiles code.

**Reply by systemed** [reply, marked as answer]
If all placenames are missing then it might be a mismatch between the name attribute that tilemaker is writing, and the one that your stylesheet is using. Unfortunately the OMT spec is really ambiguous on this and different, nominally OMT-based stylesheets appear to expect different attributes. Have a look at the comments at the start of process-openmaptiles.lua - there is special handling for this and you can easily change it to write to a particular attribute.

**Reply by diehl** [reply]
Roger that - thanks @systemed!
